File: frontend/src/model/Manifest.ts
import {Consignment} from '@/model/Consignment'

export class Manifest {
    id: string
    manifestNumber: string
    goodsPickupStatus: boolean
    userClientId: number
    readyTime: string
    closingTime: string
    dataStatus: string
    consignments: Consignment []

    constructor (id: string = null, manifestNumber: string = null, goodsPickupStatus: boolean = null, userClientId: number = null,
                 readyTime: string = null, closingTime: string = null, dataStatus: string = null, consignments: Consignment [] = null) {

        this.id = id
        this.manifestNumber= manifestNumber
        this.goodsPickupStatus = goodsPickupStatus
        this.userClientId = userClientId
        this.readyTime = readyTime
        this.closingTime = closingTime
        this.dataStatus = dataStatus
        this.consignments = consignments
    }
}

File: backend/src/main/java/com/freightmate/harbour/model/Manifest.java
package com.freightmate.harbour.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import javax.persistence.*;
import java.util.List;

@SuperBuilder
@AllArgsConstructor
@Entity
@Getter
@Setter
public class Manifest extends BaseEntity<Long> {

    @ManyToOne(targetEntity = UserClient.class, fetch = FetchType.LAZY)
    @JoinColumn(name = "user_client_id")
    private UserClient userClient;

    @Column(name = "user_client_id", insertable = false, updatable = false)
    private long userClientId;

    private Boolean goodsPickupStatus;
    private String manifestNumber;
    private String readyTime;
    private String closingTime;
    private String dataStatus;

    @OneToMany(targetEntity = Consignment.class)
    @JoinColumn(name = "manifestId")
    private List<Consignment> consignments;

    public Manifest() {
        this.isDeleted = false;
    }
}
